nsdb-core/src/main/scala/io/radicalbit/nsdb/post_proc/package.scala

```diff
@@ -30,32 +39,58 @@ package object post_proc {
     * @param chainedResults sequence of chained partial results.
     * @param statement the initial sql statement.
     * @param schema metric's schema.
+    * @param aggregationType aggregation type (temporal, count, sum etc.)
     * @return the final result obtained from the manipulation of the partials.
     */
   def applyOrderingWithLimit(
-      chainedResults: Future[Either[SelectStatementFailed, Seq[Bit]]],
+      chainedResults: Seq[Bit],
       statement: SelectSQLStatement,
-      schema: Schema)(implicit ec: ExecutionContext): Future[Either[SelectStatementFailed, Seq[Bit]]] = {
-    chainedResults.map {
-      case Right(seq) =>
-        statement.groupBy match {
-          case Some(TemporalGroupByAggregation(_, _, _)) =>
-            val sortedResults = seq.sortBy(_.timestamp)
-            statement.limit.map(_.value).map(v => Right(sortedResults.takeRight(v))) getOrElse Right(sortedResults)
-          case _ =>
-            val sortedResults = statement.order.map { order =>
-              val o = schema.fieldsMap(order.dimension).indexType.ord
-              implicit val ord: Ordering[Any] =
-                if (statement.order.get.isInstanceOf[DescOrderOperator]) o.reverse
-                else o
-              seq.sortBy(_.fields(statement.order.get.dimension)._1.rawValue)
-            } getOrElse seq
-            statement.limit.map(_.value).map(v => Right(sortedResults.take(v))) getOrElse Right(sortedResults)
-        }
-      case l @ Left(_) => l
+      schema: Schema,
+      aggregationType: Option[InternalAggregation] = None)(implicit ec: ExecutionContext): Seq[Bit] = {
+    val sortedResults = aggregationType match {
+      case Some(_: InternalTemporalAggregation) =>
+        val sortedResults = chainedResults.sortBy(_.timestamp)
```

Comment:
suspicious shadow?

Reply:
nice catch. It was partially deliberated, anyway, i'll fix it.